plugins/notifications/src/components/NotificationsTable/NotificationsTable.tsx

```diff
@@ -117,6 +123,31 @@ export const NotificationsTable = ({
     [notificationsApi, onUpdate],
   );
 
+  const onMarkAllRead = React.useCallback(() => {
+    confirm({
+      title: 'Are you sure?',
+      description: (
+        <>
+          Mark <b>all</b> notifications as <b>read</b>.
+        </>
+      ),
+      confirmationText: 'Mark All',
+    })
+      .then(async () => {
+        const ids = (
+          await notificationsApi.getNotifications({ read: false })
+        ).notifications?.map(notification => notification.id);
+
+        return notificationsApi
+          .updateNotifications({
+            ids,
+            read: true,
+          })
+          .then(onUpdate);
+      })
+      .catch(noop);
```

Comment:
Maybe instead noop, use the alertApi to show user an error?

Reply:
done